src/cart/crt_context.c

```diff
@@ -975,6 +976,8 @@ crt_context_req_untrack(struct crt_rpc_priv *rpc_priv)
 		D_MUTEX_UNLOCK(&crt_ctx->cc_mutex);
 	}
 
+	rpc_priv->crp_ctx_tracked = 0;
+
 	/* decref corresponding to addref in crt_context_req_track */
 	RPC_DECREF(rpc_priv);
```

Comment:
A belt-and-braces approach would involve setting rpc_priv to NULL here.

Reply:
Not sure what setting rpc_priv = NULL would have solved/prevented as rpc_priv used to be re-assigned to traverse the wait q list later on anyway